client/data/site-settings/use-update-site-settings-mutation.js

```diff
@@ -0,0 +1,26 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useCallback } from 'react';
+import wp from 'calypso/lib/wp';
+
+function useUpdateSiteSettingsMutation( siteId, queryOptions = {} ) {
+	const queryClient = useQueryClient();
+	const mutation = useMutation( {
+		mutationFn: ( { settings } ) =>
+			wp.req.post( '/sites/' + siteId + '/settings', { apiVersion: '1.4' }, settings ),
+		...queryOptions,
+		onSuccess( ...args ) {
+			queryClient.invalidateQueries( {
+				queryKey: [ 'site-settings', siteId ],
+			} );
+			queryOptions.onSuccess?.( ...args );
```

Comment:
We might need to invalidate a few more things on success:

There's some normalization of settings going on in the original Redux version:

https://github.com/Automattic/wp-calypso/blob/e1f240e1fb5a4083592a787b3e6a4f4c9f553692/client/state/site-settings/actions.js#L100-L101